### DISCUSSION TITLE
reimportation après migration

### DISCUSSION THREAD

**Discussion by ChevaucheurDeVague** [dbarzin/mercator#753] [Q&A] [answered]
bonjour,

j'ai fraichement réinstallé Mercator sur une VM ubuntu. Installation OK
j'ai réimporté une sauvegarde de 2023, mais cela me fait une erreur 500.(même après suppression de la base Mercator,création base,  seed et import)
J'ai tenté de faire un `php artisan migrate` mais j'ai une erreur directe avec 2019_12_14_000001_create_personal_access_tokens_table 
Cela m'aurait arranger de récupérer les données de la base.
y'a t'il une étape que j'ai zappé?
Sinon je vais reprendre la cartographie du début en faisant des sauvegardes fiables....
bien sur mon ancienne vm avec Mercator dessus n'est plus disponible...

**Comment by dbarzin** [comment, marked as answer]
Je soupçonne que tu as installé Mercator puis importé les données. Ce n'est pas le bon ordre.

Il faut récupérer les sources : 

    git clone https://www.github.com/dbarzin/mercator

Restaurer la DB :

   mysql mercator < backup.sql

Migrer la base de données

    php artisan migrate

et cela devrait aller.

**Comment by ChevaucheurDeVague** [comment]
OK je viens de remettre dans le bon ordre.

en partant de mon install bancale avec erreur 500 :

J'ai effacé la base 
`sudo mysql -e "drop database mercator;"`
puis recréé la base
`sudo mysql -e "CREATE DATABASE mercator CHARACTER SET utf8 COLLATE utf8_general_ci;"`
puis récupéré les sources
`git clone https://www.github.com/dbarzin/mercator`
puis importé ma sauvegarde
`sudo mysql mercator < backup_mercator2023bis.sql `
et finalement 
`sudo php artisan migrate`

et là cela fonctionne 

Merci Didier!

**Reply by dbarzin** [reply]
Super !

**Reply by ChevaucheurDeVague** [reply]
tout fonctionne mais à noter que j'ai eu une erreur a la fin de la migration :

```
 2024_06_05_071308_add_ldap_columns_to_users_table .......................................... 12ms FAIL

   Illuminate\Database\QueryException

  SQLSTATE[42S21]: Column already exists: 1060 Duplicate column name 'guid' (Connection: mysql, SQL: alter table `users` add `guid` varchar(255) null, add `domain` varchar(255) null)

  at vendor/laravel/framework/src/Illuminate/Database/Connection.php:829
    825▕                     $this->getName(), $query, $this->prepareBindings($bindings), $e
    826▕                 );
    827▕             }
    828▕
  ➜ 829▕             throw new QueryException(
    830▕                 $this->getName(), $query, $this->prepareBindings($bindings), $e
    831▕             );
    832▕         }
    833▕     }

      +9 vendor frames

  10  database/migrations/2024_06_05_071308_add_ldap_columns_to_users_table.php:26
      Illuminate\Support\Facades\Facade::__callStatic()
      +24 vendor frames

  35  artisan:37
      Illuminate\Foundation\Console\Kernel::handle()

```

**Comment by dbarzin** [comment]
Tu peux supprimer ce fichier et relancer la migration
